be/src/storage/rowset/segment_v2/column_reader.cpp

```diff
@@ -380,11 +380,10 @@ Status ColumnReader::bloom_filter(const std::vector<const vectorized::ColumnPred
 Status ColumnReader::_load_ordinal_index(bool use_page_cache, bool kept_in_memory) {
     Status st;
     if (_flags[kHasOrdinalIndexMetaPos]) {
-        auto* index_meta = _ordinal_index.meta;
+        std::unique_ptr<OrdinalIndexPB> index_meta(_ordinal_index.meta);
```

Comment:
Using `std::unique_ptr` here will violate the exception-safety, for example, if the `new` operator at the next line throws an exception, like `std::bad_alloc`, then the `_ordinal_index.meta` will be freed by the `unique_ptr`, but since you didn't set _flags[ kHasOrdinalIndexMetaPos] to false, `_ordinal_index.meta` will be freed again in the destructor of `ColumnReader`, caused double free.
I think the original code is exception safe already.

Reply:
ok, but i have different throughs about: original code is exception-safe already.  when **_ordinal_index.reader->load** cause exception， **kHasZoneMapIndexReaderPos**  will not be set true, so **_ordinal_index.reader** can not be free.

so may should modify look like ?
        std::unique_ptr<OrdinalIndexPB> index_meta(_ordinal_index.meta);
        _flags.set(kHasOrdinalIndexMetaPos, false);
        _ordinal_index.reader = new OrdinalIndexReader();
        _flags.set(kHasOrdinalIndexReaderPos, true);

Reply:
Yes, you're right.